team_project_access_integration_test.go

```diff
@@ -242,6 +353,99 @@ func TestTeamProjectAccessesUpdate(t *testing.T) {
 
 		assert.Equal(t, tpa.Access, TeamProjectAccessRead)
 	})
+
+	t.Run("with valid custom permissions attributes for all permissions", func(t *testing.T) {
+		skipUnlessBeta(t)
+		options := TeamProjectAccessUpdateOptions{
+			Access: ProjectAccess(TeamProjectAccessCustom),
+			ProjectAccess: &ProjectAccessPermissionsOptions{
+				Settings: ProjectSettingsPermission(ProjectSettingsPermissionUpdate),
+				Teams:    ProjectTeamsPermission(ProjectTeamsPermissionManage),
+			},
+			WorkspaceAccess: &WorkspaceAccessPermissionsOptions{
+				Runs:          WorkspaceRunsPermission(WorkspaceRunsPermissionPlan),
+				SentinelMocks: WorkspaceSentinelMocksPermission(WorkspaceSentinelMocksPermissionNone),
+				StateVersions: WorkspaceStateVersionsPermission(WorkspaceStateVersionsPermissionReadOutputs),
+				Variables:     WorkspaceVariablesPermission(WorkspaceVariablesPermissionRead),
+				Create:        false,
+				Locking:       false,
+				Move:          false,
+				Delete:        true,
+				RunTasks:      true,
+			},
+		}
+
+		tpa, err := client.TeamProjectAccess.Update(ctx, tpaTest.ID, options)
+		require.NoError(t, err)
+
+		assert.Equal(t, tpa.Access, TeamProjectAccessCustom)
+		assert.Equal(t, *options.ProjectAccess.Teams, tpa.ProjectAccess.ProjectTeamsPermission)
+		assert.Equal(t, *options.ProjectAccess.Settings, tpa.ProjectAccess.ProjectSettingsPermission)
+		assert.Equal(t, *options.WorkspaceAccess.Runs, tpa.WorkspaceAccess.WorkspaceRunsPermission)
+		assert.Equal(t, *options.WorkspaceAccess.SentinelMocks, tpa.WorkspaceAccess.WorkspaceSentinelMocksPermission)
+		assert.Equal(t, *options.WorkspaceAccess.StateVersions, tpa.WorkspaceAccess.WorkspaceStateVersionsPermission)
+		assert.Equal(t, *options.WorkspaceAccess.Variables, tpa.WorkspaceAccess.WorkspaceVariablesPermission)
+		assert.Equal(t, false, tpa.WorkspaceAccess.WorkspaceCreatePermission)
+		assert.Equal(t, false, tpa.WorkspaceAccess.WorkspaceLockingPermission)
+		assert.Equal(t, false, tpa.WorkspaceAccess.WorkspaceMovePermission)
+		assert.Equal(t, true, tpa.WorkspaceAccess.WorkspaceDeletePermission)
+		assert.Equal(t, true, tpa.WorkspaceAccess.WorkspaceRunTasksPermission)
+	})
+
+	t.Run("with valid custom permissions attributes for some permissions", func(t *testing.T) {
+		// create tpaCustomTest to verify unupdated attributes stay the same for custom permissions
+		// because going from admin to read to custom changes the values of all custom permissions
+		tm2Test, tm2TestCleanup := createTeam(t, client, orgTest)
+		defer tm2TestCleanup()
+
+		TpaOptions := TeamProjectAccessAddOptions{
+			Access:  *ProjectAccess(TeamProjectAccessCustom),
+			Team:    tm2Test,
+			Project: pTest,
+		}
+
+		tpaCustomTest, err := client.TeamProjectAccess.Add(ctx, TpaOptions)
```

Comment:
Slight capitalization nit:

```suggestion
		tpaOptions := TeamProjectAccessAddOptions{
			Access:  *ProjectAccess(TeamProjectAccessCustom),
			Team:    tm2Test,
			Project: pTest,
		}

		tpaCustomTest, err := client.TeamProjectAccess.Add(ctx, tpaOptions)
```